tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodAttributeExecuteIntegration.cs

```diff
@@ -56,80 +74,99 @@ public static class TestMethodAttributeExecuteIntegration
-                    {
-                        case UnitTestOutcome.Error:
-                        case UnitTestOutcome.Failed:
-                        case UnitTestOutcome.Timeout:
-                            test.SetErrorInfo(errorType, errorMessage, errorStackTrace);
-                            test.Close(TestStatus.Fail);
-                            break;
-                        case UnitTestOutcome.Inconclusive:
-                        case UnitTestOutcome.NotRunnable:
-                            if (exception is not null)
+                            var testExceptionType = testException.GetType();
+                            var testExceptionName = testExceptionType.Name;
+                            if (testExceptionName != "UnitTestAssertException" && testExceptionName != "AssertInconclusiveException")
                             {
-                                test.SetErrorInfo(exception);
-                                test.Close(TestStatus.Fail);
+                                test.SetErrorInfo(testException);
                             }
                             else
                             {
-                                test.Close(TestStatus.Skip, TimeSpan.Zero, errorMessage);
+                                test.SetErrorInfo(testExceptionType.FullName ?? "Error", testException.Message, testException.ToString());
                             }
+                        }
 
-                            break;
-                        case UnitTestOutcome.Passed:
-                            if (exception is not null)
-                            {
-                                test.SetErrorInfo(exception);
-                                test.Close(TestStatus.Fail);
-                            }
-                            else
+                        if (!string.IsNullOrEmpty(testResult.DisplayName) && test.Name != testResult.DisplayName)
+                        {
+                            test.SetName(testResult.DisplayName);
+                            MsTestIntegration.UpdateTestParameters(test, testMethodState.TestMethod, testResult.DisplayName);
+                        }
+
+                        if (exception is not null)
+                        {
+                            test.Close(TestStatus.Fail);
+                        }
+                        else
+                        {
+                            switch (testResult.Outcome)
                             {
-                                test.Close(TestStatus.Pass);
+                                case UnitTestOutcome.Error or UnitTestOutcome.Failed or UnitTestOutcome.Timeout:
+                                    test.Close(TestStatus.Fail);
+                                    break;
+                                case UnitTestOutcome.Inconclusive or UnitTestOutcome.NotRunnable:
+                                    test.Close(TestStatus.Skip, TimeSpan.Zero, testException?.Message ?? string.Empty);
```

Comment:
Just confirming you changed from sometimes `Fail` -> always `Skip` here. I assume that's intentional, just confirming.

Reply:
nope, the failure is handled before the switch, when `exception` is not null we fail the test.